kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue

```diff
@@ -275,38 +273,22 @@
           }
         }
       },
-      /** Returns a promise which will get the progress for a set of nodes and map it back onto
-       *  the nodes, which are then set to this.viewResourcesContents.
-       *  @modifies this.viewResourcesContents - Sets it to the value of nodes, mapped with
-       *  progress, if found
-       *  @returns {Promise} - Just returns a meaningless Promise in case you want to chain
-       *  synchronously
+      /** Calls useContentNodeProgress#fetchContentNodeProgress() with the given nodes
+       *  in order to populate the contentNodeProgressMap for use elsewhere (AlsoInThis)
+       *  @modifies this.contentNodeProgressMap (indirectly)
        */
-      progressPromise(nodes) {
-        // Avoid unnecessary fetches, set values, return empty Promise
-        if (!this.$store.getters.isUserLoggedIn) {
-          nodes.map(node => (node.progress = 0));
-          this.viewResourcesContents = nodes;
-          return Promise.resolve();
-        }
+      mapSidePanelContentProgress(nodes) {
         const getParams = {
-          content_ids: nodes.map(node => node.content_id),
+          ids: nodes.map(node => node.id),
         };
-        return ContentNodeProgressResource.fetchCollection({ getParams }).then(progresses => {
-          this.viewResourcesContents = nodes
-            .map(node => {
-              const matchingProgress = progresses.find(p => p.content_id === node.content_id) || {
-                progress: 0,
-              };
-              node.progress = matchingProgress.progress;
-              return node;
-            })
-            .filter(node => node.content_id !== this.content.content_id);
-        });
+        return this.fetchContentNodeProgress(getParams);
       },
       /**
-       * Prepares a list of content nodes which are also in the same lesson as this.lessonId
-       * @modifies this.viewResourcesContents - Sets it to the progress-mapped nodes
+       * Retrieves the lesson, then populates contentNodeProgressMap for the resulting
+       * nodes, then sets those nodes (sans this.content) to viewResourcesContents
+       * @modifies this.viewResourcesContents - Assigned the content nodes retrieved
+       * @modifies useContentNodeProgress.contentNodeProgressMap (indirectly) if the user
+       * is logged in
        */
       fetchLessonSiblings() {
         LessonResource.fetchModel({ id: this.lessonId }).then(lesson => {
```

Comment:
Can just call `this.fetchContentNodeProgress({ lesson_id: this.lessonId });` here in parallel to the lesson fetch to avoid chaining (with the isUserLoggedIn check).